SmartDeviceLink_Example/PerformInteractionManager.m

```diff
@@ -7,48 +7,87 @@
-            return;
-        }
+- (void)showWithTriggerSource:(SDLTriggerSource)source {
+    [self.manager.screenManager presentSearchableChoiceSet:self.choiceSet mode:[self modeForTriggerSource:source] withKeyboardDelegate:self];
+}
 
-        if ([response.resultCode isEqualToEnum:SDLResultTimedOut]) {
-            SDLLogD(@"The perform interaction choice set menu timed out before the user could select an item");
-            [manager sendRequest:[[SDLSpeak alloc] initWithTTS:TTSYouMissed]];
-        } else if ([response.resultCode isEqualToEnum:SDLResultSuccess]) {
-            SDLLogD(@"The user selected an item in the perform interaction choice set menu");
-            [manager sendRequest:[[SDLSpeak alloc] initWithTTS:TTSGoodJob]];
-        }
-    }];
+- (SDLChoiceSet *)choiceSet {
+    return [[SDLChoiceSet alloc] initWithTitle:PICSInitialPrompt delegate:self layout:SDLChoiceSetLayoutList timeout:10 initialPromptString:PICSInitialPrompt timeoutPromptString:PICSTimeoutPrompt helpPromptString:PICSHelpPrompt vrHelpList:nil choices:self.cells];
 }
 
-+ (NSArray<SDLChoice *> *)sdlex_createChoiceSet {
-    SDLChoice *firstChoice = [[SDLChoice alloc] initWithId:1 menuName:PICSFirstChoice vrCommands:@[PICSFirstChoice]];
-    SDLChoice *secondChoice = [[SDLChoice alloc] initWithId:2 menuName:PICSSecondChoice vrCommands:@[PICSSecondChoice]];
-    SDLChoice *thirdChoice = [[SDLChoice alloc] initWithId:3 menuName:PICSThirdChoice vrCommands:@[PICSThirdChoice]];
+- (NSArray<SDLChoiceCell *> *)cells {
+    SDLChoiceCell *firstChoice = [[SDLChoiceCell alloc] initWithText:PICSFirstChoice];
+    SDLChoiceCell *secondChoice = [[SDLChoiceCell alloc] initWithText:PICSSecondChoice];
+    SDLChoiceCell *thirdChoice = [[SDLChoiceCell alloc] initWithText:PICSThirdChoice];
+
     return @[firstChoice, secondChoice, thirdChoice];
 }
 
-+ (SDLPerformInteraction *)sdlex_createPerformInteractionWithTriggerSource:(SDLTriggerSource)triggerSource {
-    SDLInteractionMode interactionMode = [triggerSource isEqualToEnum:SDLTriggerSourceVoiceRecognition] ? SDLInteractionModeVoiceRecognitionOnly : SDLInteractionModeManualOnly;
-    SDLPerformInteraction *performInteraction = [[SDLPerformInteraction alloc] initWithInitialPrompt:PICSInitialPrompt initialText:PICSInitialText interactionChoiceSetIDList:@[@(ChoiceSetId)] helpPrompt:PICSHelpPrompt timeoutPrompt:PICSTimeoutPrompt interactionMode:interactionMode timeout:10000];
-    performInteraction.interactionLayout = SDLLayoutModeListOnly;
-    return performInteraction;
+- (SDLInteractionMode)modeForTriggerSource:(SDLTriggerSource)source {
+    return ([source isEqualToEnum:SDLTriggerSourceMenu] ? SDLInteractionModeManualOnly : SDLInteractionModeVoiceRecognitionOnly);
+}
+
+#pragma mark - SDLChoiceSetDelegate
+
+- (void)choiceSet:(SDLChoiceSet *)choiceSet didSelectChoice:(SDLChoiceCell *)choice withSource:(SDLTriggerSource)source {
```

Comment:
This method needs to be updated to include the new `rowIndex` param. App crashes.

Reply:
Fixed